packages/integration_test/lib/integration_test_driver_extended.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'common.dart';
+
+import 'package:flutter_driver/flutter_driver.dart';
+
+/// Example Integration Test which can also run WebDriver action depending on
+/// the requests coming from the test methods.
+Future<void> integrationDriver() async {
+  final FlutterDriver driver = await FlutterDriver.connect();
+
+  // Test states that it's waiting on web driver commands.
+  String jsonResponse = await driver.requestData(
+      '${TestStatus.waitOnWebdriverCommand}',
+      timeout: const Duration(seconds: 10));
```

Comment:
We have to use the `request_data` command. That is what integration_test.dart is expecting/responding. Since it only carries a `message` string we can't add anything more than a "magic string".  Note that, the design review for this handshake is passed in February. We could have write e2e in a different way if this comment was requested before all those tests are started. No, changing `request_data` would be a breaking change and will effect all the other tests so I consider that refactor a big project outside of the scope of this work.

The only non-breaking change we can make is: add a json instead of a string and serialize an object. Like we did back in this PR for the response: https://github.com/flutter/plugins/pull/2593/files#diff-e8b46d9b4e394d1ff162bfb77767c911R10

Reply:
Can we drop the timeout here? We shouldn't be recommending that people use timeouts.

Reply:
I removed all the timeouts.

I removed TestStatus enum, I added an object instead of the string. I hope it's more readable now.